githooks: avoid hitting a bug in OSX grep

knz opened the pull request:
OSX grep doesn't know how to handle an alternation with an empty
operand. Work around this by including the suffix in the alternation.

Found by @petermattis 

Also this patch makes sed/awk/grep overridable, which will facilitate
troubleshooting of further issues of this kind.

Release note: None

knz committed:
githooks: make the commit message linter more flexible

This patch makes sed/awk/grep overridable, which will facilitate
troubleshooting of portability issues.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/21679)
<!-- Reviewable:end -->

knz commented:
aw I just realize this approach may be wrong. It's not a warning that's being encountered but a plain error.

knz committed:
githooks: avoid hitting a bug in OSX grep

OSX grep doesn't know how to handle an alternation with an empty
operand. Work around this by including the suffix in the alternation.

Release note: None

knz commented:
ok done PTAL
